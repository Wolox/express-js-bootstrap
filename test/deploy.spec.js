const utils = require('./helpers/utils');
const { mockCommand } = require('./helpers/mocks');
const { examplePrompts } = require('./helpers/constants');

beforeAll(() => mockCommand());

const deployOptions = [
  ['docker', { files: ['Dockerfile'], kickoffOptions: { docker: true, deployStrategy: { aws: false } } }],
  [
    'docker and aws',
    {
      files: ['Dockerfile', 'Dockerrun.aws.json', '.ebextensions/cloudwatch.config'],
      kickoffOptions: { docker: true, deployStrategy: { aws: true } }
    }
  ],
  [
    'heroku',
    { files: ['Procfile'], kickoffOptions: { docker: false, deployStrategy: { aws: false, heroku: true } } }
  ],
  [
    'aws',
    {
      files: ['.ebextensions/cloudwatch.config'],
      kickoffOptions: { docker: true, deployStrategy: { aws: true } }
    }
  ]
];

describe.each(deployOptions)('Deploy with %s', (deployOption, { files, kickoffOptions }) => {
  beforeAll(() =>
    utils.runKickoff({
      ...examplePrompts,
      ...kickoffOptions,
      projectName: 'DeployProject'
    })
  );

  test(`creates files for ${deployOption}`, () => {
    utils.checkExistentFiles(files, 'DeployProject');
  });

  test.each(files)('creates expected %s', file => {
    expect(utils.getFileContent(`DeployProject/${file}`)).toMatchSnapshot();
  });
});
