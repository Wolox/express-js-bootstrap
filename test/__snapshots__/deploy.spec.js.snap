// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deploy with aws creates expected .ebextensions/cloudwatch.config 1`] = `
"packages:
  yum:
    perl-DateTime: []
    perl-Sys-Syslog: []
    perl-LWP-Protocol-https: []
    perl-Switch: []
    perl-URI: []
    perl-Bundle-LWP: []
sources: 
  /opt/cloudwatch: https://aws-cloudwatch.s3.amazonaws.com/downloads/CloudWatchMonitoringScripts-1.2.1.zip

container_commands:
  01-setupcron:
    command: |
      echo '*/5 * * * * root perl /opt/cloudwatch/aws-scripts-mon/mon-put-instance-data.pl \`{\\"Fn::GetOptionSetting\\" : { \\"OptionName\\" : \\"CloudWatchMetrics\\", \\"DefaultValue\\" : \\"--mem-util --disk-space-util --disk-path=/\\" }}\` >> /var/log/cwpump.log 2>&1' > /etc/cron.d/cwpump
  02-changeperm:
    command: chmod 644 /etc/cron.d/cwpump
  03-changeperm:
    command: chmod u+x /opt/cloudwatch/aws-scripts-mon/mon-put-instance-data.pl
option_settings:
  \\"aws:autoscaling:launchconfiguration\\" :
    IamInstanceProfile : \\"aws-elasticbeanstalk-ec2-role\\"
  \\"aws:elasticbeanstalk:customoption\\" :
    CloudWatchMetrics : \\"--mem-util --mem-used --mem-avail --disk-space-util --disk-space-used --disk-space-avail --disk-path=/ --auto-scaling\\"
"
`;

exports[`Deploy with docker and aws creates expected .ebextensions/cloudwatch.config 1`] = `
"packages:
  yum:
    perl-DateTime: []
    perl-Sys-Syslog: []
    perl-LWP-Protocol-https: []
    perl-Switch: []
    perl-URI: []
    perl-Bundle-LWP: []
sources: 
  /opt/cloudwatch: https://aws-cloudwatch.s3.amazonaws.com/downloads/CloudWatchMonitoringScripts-1.2.1.zip

container_commands:
  01-setupcron:
    command: |
      echo '*/5 * * * * root perl /opt/cloudwatch/aws-scripts-mon/mon-put-instance-data.pl \`{\\"Fn::GetOptionSetting\\" : { \\"OptionName\\" : \\"CloudWatchMetrics\\", \\"DefaultValue\\" : \\"--mem-util --disk-space-util --disk-path=/\\" }}\` >> /var/log/cwpump.log 2>&1' > /etc/cron.d/cwpump
  02-changeperm:
    command: chmod 644 /etc/cron.d/cwpump
  03-changeperm:
    command: chmod u+x /opt/cloudwatch/aws-scripts-mon/mon-put-instance-data.pl
option_settings:
  \\"aws:autoscaling:launchconfiguration\\" :
    IamInstanceProfile : \\"aws-elasticbeanstalk-ec2-role\\"
  \\"aws:elasticbeanstalk:customoption\\" :
    CloudWatchMetrics : \\"--mem-util --mem-used --mem-avail --disk-space-util --disk-space-used --disk-space-avail --disk-path=/ --auto-scaling\\"
"
`;

exports[`Deploy with docker and aws creates expected Dockerfile 1`] = `
"FROM node:8.9.12

WORKDIR /home/node/app

COPY package.json .
COPY package-lock.json .
COPY .nvmrc .

RUN npm install

COPY . .

EXPOSE 8080
ENV NODE_ENV production
CMD [\\"node\\", \\"server.js\\"]
"
`;

exports[`Deploy with docker and aws creates expected Dockerrun.aws.json 1`] = `
"{
  \\"AWSEBDockerrunVersion\\": \\"1\\",
  \\"Ports\\": [
      {
          \\"ContainerPort\\": \\"8080\\"
      }
  ],
  \\"Volumes\\": [],
  \\"Logging\\": \\"/home/node/app/app/logger/logs\\"
}
"
`;

exports[`Deploy with docker creates expected Dockerfile 1`] = `
"FROM node:8.9.12

WORKDIR /home/node/app

COPY package.json .
COPY package-lock.json .
COPY .nvmrc .

RUN npm install

COPY . .

EXPOSE 8080
ENV NODE_ENV production
CMD [\\"node\\", \\"server.js\\"]
"
`;

exports[`Deploy with heroku creates expected Procfile 1`] = `
"web: node server.js
"
`;
